src/ai/backend/client/cli/admin/domain.py

```diff
@@ -90,11 +91,17 @@ def add(name, description, inactive, total_resource_slots,
         if not data['ok']:
             print_fail('Domain creation has failed: {0}'.format(data['msg']))
             sys.exit(1)
-        item = data['domain']
-        print('Domain name {0} is created.'.format(item['name']))
+        ctx.output.print_mutation_result(
+            data,
+            item_name='domain',
+            add_info={
+                'method': sys._getframe().f_code.co_name,
```

Comment:
This is too much hacky, because it uses a CPython-specific private API. (CPython is just one of many Python language implementations, though it is the reference implementation.)

To extract the command name, we could use Click's API instead.

```python
@click.pass_context
...
def add(click_ctx: click.Context, ...):
    ...
    print(click_ctx.command.name)
```

There are many more rich information about the current command in [the Click's context object](https://click.palletsprojects.com/en/8.0.x/api/#context).

It is better to use publicly available APIs whenever possible.

Reply:
BTW, since the purpose of JSON output is to mechanically automate user-side scripting and the caller already knows what command it called because it called, I don't think this is a strictly required information to include in every mutation's JSON output.

Please justify the scenarios where this field is useful or remove it.

Reply:
I removed those function name field.
I simply wrote it because CLI shows what function is called with --output=console option.